scripts/main.js

```diff
@@ -0,0 +1,48 @@
+const meuCabecalho = document.getElementById('initial-text');
+meuCabecalho.textContent = 'Github Profile Search';
+
+const meuBotao = document.querySelector('button');
+meuBotao.addEventListener("click", getUser);
+
+
+function getUser() {  
+    document.getElementById('list-rep').innerHTML = '';
+    var gitUser = document.querySelector('input')
+    fetch(`https://api.github.com/users/${gitUser.value}`)  //Search for github user 
+    .then(response => {
+      return response.json() //Converting the response to a JSON object
+    }) 
+    .then(function(data) {
+      if(data.message){
+          console.log('User Profile Not Found')
+      }else{
+          console.log(data)
+          document.getElementById('res').innerHTML = `${data.name}`
+      }
+      return fetch(`https://api.github.com/users/${gitUser.value}/repos`) // Search for Repositories
+    })
+    .then(response => {
+      return response.json()  //Converting the response to a JSON object
+    })
+    .then(data =>{
+          console.log(data)
+          document.getElementById('rep').innerHTML = putRepo(data) // Display repositories list
+    })
+    
+    
+    .catch(function(error) {
+      console.log(error)
+    });
+}
+
+function putRepo(data) {
+  var listDiv = document.getElementById("list-rep");
+  var ul=document.createElement('ul');
+  for (var i = 0; i < data.length; ++i) {
```

Comment:
Evitar utilizar var, sempre que possível utilizar let ou const.

Existem outrar maneiras de escrever este for que eu gostaria de compartilhar:
For in: https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/for...in
```javascript
for (let index in data) {
```

For of: pode evitar ter que pegar o objeto com data[index]; https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/for...of
```javascript
for (let object of data) {
```

forEach: caso seja necessário ter as duas variáveis prontas; https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
```javascript
data.forEach((object, index)=>{})
```

como você apenas utiliza data[i] sugiro utilizar o for of